README.md

```diff
@@ -88,3 +88,61 @@ If the snapshot tests fail, it's generally pretty easy to tell whether it's happ
+## Semantic Release
+
+Paragon uses the [`semantic-release` package](https://github.com/semantic-release/semantic-release) to automate it's release process (creating Git tags, creating GitHub releases, and publishing to NPM).
+
+### Commit Messages
+
+[`semantic-release` analyzes commit messages to determine whether to create a `major`, `minor`, or `patch` release](https://github.com/semantic-release/semantic-release#default-commit-message-format) (or to skip a release).
+Paragon currently uses [the default commit analyzer release rules](https://github.com/semantic-release/commit-analyzer/blob/master/lib/default-release-rules.js#L8-L11) which means that there are **4** commit types that will trigger a release:
+1. `feat` (`minor` release)
+2. `fix` (`patch` release)
+3. `perf` (`patch` release)
+4. `breaking` (`major` release)
+
+#### `Angular` Commit Message Convention
+
+Paragon currently uses [the `Angular` commit message convention](https://gist.github.com/stephenparish/9941e89d80e2bc58a153). As documented in the previously linked gist, a commit that follows the `Angular` commit message convention has four parts:
+1. A `type` - is this commit a `feat`, `fix`, `chore`, `doc`, etc.? There is a set of `type` values to choose from, but again, **only the `feat`, `fix`, `perf`, and `breaking`** `type` values will trigger a release.
+2. A `scope` - what is this commit impacting? Did you fix a bug in the `Hyperlink` component? Did you add a new feature to the `RadioButtonGroup` component? Currently, the `scope` must be lower-case and `-` separated though switching this to being `camelCase` is currently being investigated.
+3. A `subject` - provide a short description of _what_ your change is.
+  * use imperative, present tense language (so `change` not `changed` or `changes`)
+  * don't capitalize the first letter
+  * don't add a period (`.`) at the end
+  * there is a 50 character limit
+4. A `body` (optional) - add more detail about your change
+5. A `footer` (optional)
+  * > All breaking changes have to be mentioned in footer with the description of the change, justification and migration notes - [`Angular` commit message specification](https://gist.github.com/stephenparish/9941e89d80e2bc58a153#breaking-changes)
+
+#### Using `commitlint` in Paragon
+
+Paragon uses the [`commitlint`](https://github.com/marionebl/commitlint) package to lint commit messages. Paragon currently has [a `commit message` hook](https://github.com/edx/paragon/blob/master/package.json#L20) that runs `commitlint`'s, well, commit linting, on the given commit message using the configuration specified in [`commitlint.config.js`](https://github.com/edx/paragon/blob/master/commitlint.config.js). If a commit message fails linting, the commit associated with that message does not end up getting committed.
+
+`commitlint` also comes with [a helpful CLI](https://github.com/marionebl/commitlint/tree/master/@commitlint/cli) that walks one through the entire semantic commit process.
```

Comment:
Quick question. This CLI is invoked via `npm run gc`, right? Should we document that anywhere? Actually, now that I look at package.json, it looks like there's a `commitmsg` command. Is that the CLI command?

Reply:
lol I can add that note about `npm run gc`

Reply:
`commitmsg` actually refers to [the `husky` `commitmsg` hook](https://github.com/typicode/husky/blob/master/HOOKS.md).